packages/internals/support/src/entities/component.js

```diff
@@ -64,6 +51,25 @@ class Component extends Entity {
       get: variantsGet,
       set: variantsSet
     });
+
+    Object.defineProperty(this, 'config', {
+      enumerable: true,
+      get: configGet,
+      set: configSet
+    });
+  }
+
+  get(path, fallback) {
+    fallback = get(this.config, path, fallback);
+    const result = get(_data.get(this), path, fallback);
+    return isObjectLike(result) ? cloneDeep(result) : result;
+  }
+
+  set(path, value) {
+    assert.string(path, 'Component.set - `path` argument must be a string [path-invalid]');
+    // TODO: extract shared caching code with config: this.removeFromCache(path);
+    set(_data.get(this), path, cloneDeep(value));
```

Comment:
I believe we do, we'd want to break the link to any assigned objects both coming and going, in case something else were to modify the original passed-in value after being either assigned or returned.